iosApp/iosApp/Pages/Map/HomeMapViewHandlerExtension.swift

```diff
@@ -64,6 +64,7 @@ extension HomeMapView {
             joinVehiclesChannel(routeId: routeId, directionId: directionId)
         } else {
             leaveVehiclesChannel()
+            vehiclesData = []
```

Comment:
Because the vehicles channel was being disconnected and reconnected every time you selected another departure, which was leading to all of the vehicles flickering in and out every time. It would probably make sense to instead not disconnect and reconnect, that can probably happen in a follow up.